diff --git a/mythtv/external/FFmpeg/libavcodec/mpegvideo.c b/mythtv/external/FFmpeg/libavcodec/mpegvideo.c
index 7240779..9225270 100644
--- a/mythtv/external/FFmpeg/libavcodec/mpegvideo.c
+++ b/mythtv/external/FFmpeg/libavcodec/mpegvideo.c
@@ -999,7 +999,7 @@ int MPV_frame_start(MpegEncContext *s, AVCodecContext *avctx)
 
     if(s->codec_id != CODEC_ID_H264){
         if((s->last_picture_ptr==NULL || s->last_picture_ptr->data[0]==NULL) && s->pict_type!=FF_I_TYPE){
-            av_log(avctx, AV_LOG_ERROR, "warning: first frame is no keyframe\n");
+          /* av_log(avctx, AV_LOG_ERROR, "warning: first frame is no keyframe\n"); */
             /* Allocate a dummy frame */
             i= ff_find_unused_picture(s, 0);
             s->last_picture_ptr= &s->picture[i];
diff --git a/mythtv/libs/libmythtv/avformatdecoder.cpp b/mythtv/libs/libmythtv/avformatdecoder.cpp
index 23719d1..d5904cf 100644
--- a/mythtv/libs/libmythtv/avformatdecoder.cpp
+++ b/mythtv/libs/libmythtv/avformatdecoder.cpp
@@ -1431,8 +1431,8 @@ void AvFormatDecoder::ScanATSCCaptionStreams(int av_index)
     // Figure out languages of ATSC captions
     if (!ic->cur_pmt_sect)
     {
-        VERBOSE(VB_IMPORTANT, LOC_WARN +
-                "ScanATSCCaptionStreams() called with no PMT");
+      /* VERBOSE(VB_IMPORTANT, LOC_WARN +
+         "ScanATSCCaptionStreams() called with no PMT"); */
         return;
     }
 
